### DISCUSSION TITLE
SRGSSR API Client refactor

### DISCUSSION THREAD

**Discussion by joachimBurket** [ManBehindMooN/kodi_plugin_video_srgssr_ch_replay#11] [Ideas]
Hi there,

I'm thinking about refactoring the API Client in a class in its own python file. What do you think about it?

**Comment by ManBehindMooN** [comment]
As the code base is growing (400+ lines) it totally makes sense to move all SRGSSR API functionality into its own file.

**Comment by ManBehindMooN** [comment]
@joachimBurket I created a feature/3.0.4 branch. If you have time, just create a PR on that branch.

**Comment by joachimBurket** [comment]
Okay thanks, I'm currently working on it. The refactor overflows a bit on the plugin itself also :sweat_smile: .

**Comment by joachimBurket** [comment]
I'm almost done with a v1 of the refactor, but I have some interrogations about the authentication token for akamaihd in the `play_episode` method.
Is it related to the DRMs? I searched some shows but couldn't find any with `akamaihd`.

**Reply by ManBehindMooN** [reply]
Yeah, this akamai is responsible for IP blocking and DRM for some TV shows. Here is an example: https://il.srgssr.ch/integrationlayer/2.0/mediaComposition/byUrn/urn:srf:video:5f76d203-d4a8-450d-b575-fe1fbab26a67.json. On the SRF business unit you should find some examples. Do not remove the adamai token logic. This logic is probably not the best implementation because not every TV show needs a token only the ones that have a tokenType: "AKAMAI" as in the above example and the logic always gets a token when akamaihd is in the URL but I think is not worthwhile to refactor this part because SRG SSR will probably remove akamai and then we re-write this part from scratch.
